include/index.h

```diff
@@ -49,20 +49,14 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
      **************************************************************************/
 
   public:
-    // Constructor for Bulk operations and for creating the index object solely
-    // for loading a prexisting index.
-    DISKANN_DLLEXPORT Index(Metric m, const size_t dim, const size_t max_points = 1, const bool dynamic_index = false,
+    // For internal use - uses new constructor internally
+    DISKANN_DLLEXPORT Index(Metric m, const size_t dim, const size_t max_points,
+                            const std::shared_ptr<IndexWriteParameters> &indexParameters,
+                            const std::shared_ptr<IndexSearchParams> &indexSearchParams,
+                            const size_t num_frozen_pts = 0, const bool dynamic_index = false,
                             const bool enable_tags = false, const bool concurrent_consolidate = false,
```

Comment:
should we consolidate some of these params in write and search params?

Reply:
Its already done as you can see with Write Params (used when writing / building index) and search params.